drivers/mtd/gd55.c

```diff
@@ -0,0 +1,2065 @@
+
+ssize_t gd55_read(FAR struct mtd_dev_s *dev, off_t offset, size_t nbytes,
+                  FAR uint8_t *buffer)
+{
+  int                   ret;
+  FAR struct gd55_dev_s *priv = (FAR struct gd55_dev_s *)dev;
+
+  finfo("offset: %08lx nbytes: %d\n", (long)offset, (int)nbytes);
+
+  /* Lock the QuadSPI bus and select this FLASH part */
+
+  gd55_lock(priv);
+  ret = gd55_read_bytes(priv, buffer, offset, nbytes);
+  gd55_unlock(priv);
+
+  if (ret < 0)
+    {
+      ferr("ERROR: gd55_read_bytes returned: %d\n", ret);
+      return (ssize_t)ret;
+    }
+
+  finfo("return nbytes: %d\n", (int)nbytes);
+  return (ssize_t)nbytes;
+}
+
+/****************************************************************************
+ * Name: gd55_ioctl
+ *
+ * Description:
+ *   IOCTLS relating to the GD55 mtd device
+ *
+ * Input Parameters:
+ *   dev        - a reference to the device structure
+ *   cmd        - ioctl command
+ *   arg        - ioctl argument
+ *
+ * Returned Value:
+ *   Success (OK) or fail (negated error code)
+ ****************************************************************************/
+
+int gd55_ioctl(FAR struct mtd_dev_s *dev, int cmd, unsigned long arg)
+{
+  FAR struct gd55_dev_s *priv = (FAR struct gd55_dev_s *)dev;
+  int                    ret  = -EINVAL;
+
+  finfo("cmd: %d\n", cmd);
+
+  switch (cmd)
+    {
+      case MTDIOC_GEOMETRY:
+        {
+          FAR struct mtd_geometry_s *geo =
+            (FAR struct mtd_geometry_s *)((uintptr_t)arg);
+
+          if (geo)
+            {
+              memset(geo, 0, sizeof(*geo));
+
+              /* Populate the geometry structure with information need to
+               * know the capacity and how to access the device.
+               *
+               * NOTE:
+               * that the device is treated as though it where just an
+               * array of fixed size blocks.  That is most likely not true,
+               * but the client will expect the device logic to do whatever
+               * is necessary to make it appear so.
+               */
+
+#ifdef CONFIG_MTD_GD55_SECTOR512
+              geo->blocksize    = GD55_SECTOR512_SIZE;
+              geo->erasesize    = GD55_SECTOR512_SIZE;
+              geo->neraseblocks = priv->nsectors <<
+                                  (GD55_SECTOR_SHIFT -
+                                   GD55_SECTOR512_SHIFT);
+#else
+              geo->blocksize    = GD55_PAGE_SIZE;
+              geo->erasesize    = GD55_SECTOR_SIZE;
+              geo->neraseblocks = priv->nsectors;
+#endif
+              ret               = OK;
+
+              finfo("blocksize: %" PRId32
+                    " erasesize: %" PRId32
+                    " neraseblocks: %" PRId32 "\n",
+                    geo->blocksize, geo->erasesize, geo->neraseblocks);
+            }
+        }
+        break;
+
+      case BIOC_PARTINFO:
+        {
+          FAR struct partition_info_s *info =
+            (FAR struct partition_info_s *)arg;
+          if (info != NULL)
```

Comment:
add blank line before this line

Reply:
Agreed - should checkpatch.sh not have picked this up?

Reply:
done